docker/src/main/java/brooklyn/entity/container/docker/registry/DockerRegistryImpl.java

```diff
@@ -0,0 +1,109 @@
+package brooklyn.entity.container.docker.registry;
+
+
+import static org.apache.brooklyn.util.ssh.BashCommands.chain;
+
+import javax.naming.directory.AttributeInUseException;
+
+import java.net.URI;
+import java.util.List;
+
+import com.google.common.base.Functions;
+import com.google.common.collect.ImmutableList;
+import com.google.common.net.HostAndPort;
+
+import org.apache.brooklyn.core.entity.Attributes;
+import org.apache.brooklyn.core.location.access.BrooklynAccessUtils;
+import org.apache.brooklyn.entity.nosql.couchbase.CouchbaseNode;
+import org.apache.brooklyn.entity.software.base.SoftwareProcess;
+import org.apache.brooklyn.feed.http.HttpFeed;
+import org.apache.brooklyn.feed.http.HttpPollConfig;
+import org.apache.brooklyn.feed.http.HttpValueFunctions;
+import org.apache.brooklyn.location.ssh.SshMachineLocation;
+import org.apache.brooklyn.util.collections.MutableMap;
+import org.apache.brooklyn.util.os.Os;
+import org.apache.brooklyn.util.ssh.BashCommands;
+import org.apache.brooklyn.util.time.Duration;
+
+import brooklyn.entity.container.DockerAttributes;
+import brooklyn.entity.container.docker.DockerHost;
+import brooklyn.entity.container.docker.DockerInfrastructure;
+import brooklyn.entity.container.docker.application.VanillaDockerApplicationImpl;
+
+public class DockerRegistryImpl extends VanillaDockerApplicationImpl implements DockerRegistry {
+
+    private volatile HttpFeed httpFeed;
```

Comment:
I marked this as  volatile as I was following the precedent set in other classes, for example, RiakNodeImpl. Can remove if you think there will be no multi-threading issues?

Not sure about adding transient, is that something you would recommend? When would this be serialised? Possibly as part of persistence?